client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts

```diff
@@ -0,0 +1,56 @@
+import { translate } from 'i18n-calypso';
+import { Task } from './types';
+
+export function getEnhancedTasks( tasks: Task[], siteSlug: string | null ) {
+	const enhancedTaskList: Task[] = [];
+	tasks &&
+		tasks.map( ( task ) => {
+			let taskData = {};
+			switch ( task.id ) {
+				case 'setup_newsletter':
+					taskData = {
+						title: translate( 'Set up Newsletter' ),
+					};
+					break;
+				case 'plan_selected':
+					taskData = {
+						title: translate( 'Free Plan' ),
+					};
+					break;
+				case 'subscribers_added':
+					taskData = {
+						title: translate( 'Add Subscribers' ),
+					};
+					break;
+				case 'first_post_published':
+					taskData = {
+						title: translate( 'Write your first post' ),
+						actionUrl: `/post/${ siteSlug }`,
+					};
+					break;
+				case 'design_selected':
+					taskData = {
+						title: translate( 'Select a design' ),
+					};
+					break;
+				case 'setup_link_in_bio':
+					taskData = {
+						title: translate( 'Setup Link in bio' ),
+					};
+					break;
+				case 'links_added':
+					taskData = {
+						title: translate( 'Add links' ),
+						actionUrl: `/site-editor/${ siteSlug }`,
+					};
+					break;
+				case 'link_in_bio_launched':
```

Comment:
Should we have this step redirect to the `https://wordpress.com/start/launch-site/domains-launch?siteSlug=[YOUR_SITE_SLUG_HERE]&source=general-settings` page?

Reply:
I'm not too familiar with the existing launch flow(s). 

@Addison-Stavlo - I recall you mentioned a flow for launching sites that we could use for this checklist item. Is this the correct URL?

Reply:
Taking from how the Launch flow currently behaves in myHome, It seems like the functionality is:
* IF on free plan / no domain - redirects through the plans and domains upsell flow to go launch the site (seems like the URL Jeremy is linking?)
* If already on a plan - the Launch button just Launches right there when you press it without any reloads or redirects.

We should probably aim for something like this. Direct them into the launch-site flow with plan/domain upsells if still on the free plan, otherwise the button just triggers site launch, dismisses the launchpad, and directs them to calypso.